### PR TITLE
Sites Dashboard: Add migration status to site header

### LINKED ISSUE #95589: Post-purchase Migration Experience: Add status to site header for pending migrations
- When a user clicks on a site in the Sites dashboard that is pending migration, display the migration status in the header.
- Disable the _Hosting settings_ tab for a migration that is pending or started.

Figma - nOWSHCjXMpB5sXHthTVdvK-fi-3949_19810

Image

### PR DESCRIPTION
**Pull request by m1r0**
Part of #95589
Based on #95497

## Proposed Changes

* When the migration status is "pending", updates the site title to "Incoming Migration", changes to icon, and adds the "migration pending" status in the site header.
* Fixes the site title being updated to "Incoming Migration" when the migration status is "started". In the design, it's updated only when the status is "pending".

## Testing Instructions

* Have a blog with the "pending" migration status.
* Go to `/sites`, click on the blog, and make sure the site heading has the "Incoming Migration" title, custom icon, and the "migration pending" status added to it.

image

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/hosting/sites/components/sites-dataviews/dataviews-fields/site-field.tsx [hunk 1]
```diff
@@ -80,8 +80,8 @@ const SiteField = ( { site, openSitePreviewPane }: Props ) => {
 		event.preventDefault();
 	};
 
-	const isMigrating = isMigrationInProgress( site );
-	const siteTitle = isMigrating ? translate( 'Incoming Migration' ) : site.title;
+	const isMigrationPending = getMigrationStatus( site ) === 'pending';
+	const siteTitle = isMigrationPending ? translate( 'Incoming Migration' ) : site.title;
```
Comment (m1r0) [inline client/hosting/sites/components/sites-dataviews/dataviews-fields/site-field.tsx:90]: This fixes the site title being updated to "Incoming Migration" when the migration status is "started". In the design, it's updated only when the status is "pending".

File: client/hosting/sites/components/sites-dataviews/sites-site-status.tsx [hunk 2]
```diff
@@ -120,13 +120,12 @@ export const SiteStatus = ( { site }: SiteStatusProps ) => {
...
 
-	const isMigrationPeding =
-		site.site_migration?.migration_status?.startsWith( 'migration-pending' );
-	const statusElement = isMigrationPeding ? (
-		<span className="sites-dataviews__migration-pending-status">{ translatedStatus }</span>
-	) : (
-		translatedStatus
-	);
+	const statusElement =
+		getMigrationStatus( site ) === 'pending' ? (
```
Comment (renatho) [inline client/hosting/sites/components/sites-dataviews/sites-site-status.tsx:124]: Good catch reusing the code here! 🎉

**Comment by renatho** [comment]
Thanks @m1r0!

There's just a small thing I noticed. The "Migration pending" badge should have `12px` there.

It seems it should be the same in the table and I didn't realize when I wrote that class. Maybe we could just add the property `font-size: rem(12px);` to the `.sites-dataviews__migration-pending-status`?

**Comment by renatho** [comment]
Thanks! Now it will just need the rebase and we're good to go.

**Comment by m1r0** [comment]
Rebased! Thanks @renatho. 🙂

**Review by renatho — APPROVED** [review]
Thanks @m1r0!
Looks good and works well! 🎉

**Pull request merged**